core/request_handler.go

```diff
@@ -276,7 +276,11 @@ func (c *Connection) msgCallback(msgID uint16, data []byte) {
 				l.Debugf("Unable to decode message: %v", err)
 			}
 		}
-		l.Errorf("Channel ID not known, ignoring the message: %T %+v", msg, msg)
+		// This can happen when a stream was closed explicitly before the last message received.
+		// It's expected and not an error.
+		if log.Level >= logrus.DebugLevel {
```

Comment:
You should have included the part above that does decoding as that is actually what can have performance hit.